Deprecate `@fedify/fedify/x/*` modules and extract them into separate packages

2chanhaeng opened the pull request:
## Summary

Deprecate `@fedify/fedify/x/*` modules and extract them into separate packages.

## Related Issue

- closes #375

## Changes

- Separated modules from `@fedify/fedify/x` into dedicated packages.
- Added JSDoc for `packages/fedify/x/*` is deprecated and will be gone at v2.
- Added docs and warnings for the changes.

## Benefits

By the separation, `@fedify/fedify` will improve modularity and reduce bundle size.

## Checklist

- [x] Did you add a changelog entry to the *CHANGES.md*?
- [x] Did you write some relevant docs about this change (if it's a new feature)?
- [x] ~~Did you write a regression test to reproduce the bug (if it's a bug fix)?~~
- [x] Did you write some tests for this change (if it's a new feature)?
- [x] Did you run `deno task test-all` on your machine?

## Additional Notes

The added package has not been released yet, so it has not been applied to the code in the example. A sub-issue is needed to modify this after the package is released.

dahlia commented on packages/cfworkers/package.json:
```diff
@@ -0,0 +1,66 @@
+{
+  "name": "@fedify/cfworkers",
+  "version": "1.9.0",
+  "description": "",
```
Shouldn't this be filled up?

dahlia commented on packages/hono/package.json:
```diff
@@ -0,0 +1,66 @@
+{
+  "name": "@fedify/hono",
+  "version": "1.9.0",
+  "description": "",
```
Shouldn't this be filled up?

dahlia commented on packages/sveltekit/package.json:
```diff
@@ -0,0 +1,67 @@
+{
+  "name": "@fedify/sveltekit",
+  "version": "1.9.0",
+  "description": "",
```
Shouldn't this be filled up?

dahlia commented on packages/cfworkers/src/mod.test.ts:
```diff
@@ -0,0 +1,79 @@
+import { delay } from "es-toolkit";
+import { WorkersKvStore, WorkersMessageQueue } from "./mod.ts";
+
+Deno.test({
```
In order to make this test runnable on Node.js and Bun, we should use [`node:test`](https://nodejs.org/api/test.html) and [`node:assert/strict`](https://nodejs.org/api/assert.html).

dahlia commented on packages/cfworkers/src/mod.test.ts:
```diff
@@ -0,0 +1,79 @@
+  },
+});
+
+Deno.test({
```
In order to make this test runnable on Node.js and Bun, we should use [`node:test`](https://nodejs.org/api/test.html) and [`node:assert/strict`](https://nodejs.org/api/assert.html).

dahlia commented on packages/denokv/package.json:
I don't think we need to publish *@fedify/denokv* as an npm package as it's only for Deno.

dahlia commented on packages/hono/src/mod.test.ts:
```diff
@@ -0,0 +1,86 @@
+  fetch(request: Request, options: any): Promise<Response>;
+}
+
+Deno.test("federation middleware", async (t) => {
```
In order to make this test runnable on Node.js and Bun, we should use [`node:test`](https://nodejs.org/api/test.html) and [`node:assert/strict`](https://nodejs.org/api/assert.html).

dahlia commented on packages/sveltekit/src/mod.test.ts:
```diff
@@ -0,0 +1,84 @@
+  fetch(request: Request, options: any): Promise<Response>;
+}
+
+Deno.test("fedifyHook", async (t) => {
```
In order to make this test runnable on Node.js and Bun, we should use [`node:test`](https://nodejs.org/api/test.html) and [`node:assert/strict`](https://nodejs.org/api/assert.html).

dahlia commented on packages/denokv/tsdown.config.ts:
I don't think we need to publish *@fedify/denokv* as an npm package as it's only for Deno.

dahlia commented on packages/hono/src/mod.ts:
```diff
@@ -0,0 +1,103 @@
+ * [Hono]: https://hono.dev/
+ *
+ * @module
+ * @since 0.6.0
```
```suggestion
 * @since 1.9.0
```

dahlia commented on packages/cfworkers/src/mod.ts:
```diff
@@ -0,0 +1,146 @@
+ * Cloudflare Workers' KV and Queues bindings, respectively.
+ *
+ * @module
+ * @since 1.6.0
```
```suggestion
 * @since 1.9.0
```

dahlia commented on packages/denokv/src/mod.ts:
```diff
@@ -0,0 +1,152 @@
+ * `MessageQueue` interface using Deno's KV store.
+ *
+ * @module
+ * @since 0.5.0
```
```suggestion
 * @since 1.9.0
```

dahlia commented on packages/sveltekit/src/mod.ts:
```diff
@@ -0,0 +1,87 @@
+ * [SvelteKit]: https://kit.svelte.dev/
+ *
+ * @module
+ * @since 1.3.0
```
```suggestion
 * @since 1.9.0
```

2chanhaeng commented:
- Replaced `@fedify/fedify/x/*` modules with `@fedify/*` packages in docs.
- Replaced links for archived repository with links for [`packages/*`](https://github.com/fedify-dev/fedify/tree/main/packages)

dahlia reviewed (dismissed):
Huge thanks!
